### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by martinholters**
With the next release bumping the minor version, this might be a good time to remove the deprecations currently in place. Of course, the cost of keeping them around is also relatively little, hence RFC. Could also decide to remove only some them, of course. Maybe depending on how long they have been around?

### CODE REVIEW

File: src/deprecated.jl [hunk 1]
```diff
@@ -1,11 +0,0 @@
-# deprecations after 0.5
-Base.@deprecate_binding ZPKFilter ZeroPoleGain
-Base.@deprecate_binding TFFilter PolynomialRatio
-Base.@deprecate_binding BiquadFilter Biquad
-Base.@deprecate_binding SOSFilter SecondOrderSections
```
Comment (martinholters) [inline src/deprecated.jl:5]: These have been deprecated in 0.6.0.

File: src/deprecated.jl [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-
-Base.@deprecate Frequencies(nreal::Int, n::Int, multiplier::Float64) FFTW.Frequencies(nreal, n, multiplier)
-Base.@deprecate fftfreq(n::Int, fs::Real=1) FFTW.fftfreq(n, fs)
-Base.@deprecate rfftfreq(n::Int, fs::Real=1) FFTW.rfftfreq(n, fs)
```
Comment (martinholters) [inline src/deprecated.jl:9]: These have been deprecated in 0.6.1.

File: src/deprecated.jl [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-
-@deprecate conv2 conv
```
Comment (martinholters) [inline src/deprecated.jl:11]: This has been deprecated in 0.6.1.

File: src/dspbase.jl [hunk 4]
```diff
@@ -795,17 +774,11 @@ The size of the output depends on the padmode keyword argument: with padmode =
 With padmode = :longest the shorter of the arguments will be padded so they are
 equal length. This gives a result with length 2*max(length(u), length(v))-1,
 with the zero-lag condition at the center.
-
-!!! warning
-    The default value of `padmode` will be changing from `:longest` to `:none`
-    in a future release of DSP. In preparation for this change, leaving
-    `padmode` unspecified is currently deprecated.
```
Comment (martinholters) [inline src/dspbase.jl:802]: This was done in 0.6.0.

File: src/unwrap.jl [hunk 5]
```diff
@@ -7,13 +7,7 @@ export unwrap, unwrap!
 
 In-place version of [`unwrap`](@ref).
 """
-function unwrap!(m::AbstractArray{T,N}; dims=nothing, kwargs...) where {T, N}
-    if dims === nothing && N != 1
-        Base.depwarn("`unwrap!(m::AbstractArray)` is deprecated, use `unwrap!(m, dims=ndims(m))` instead", :unwrap!)
```
Comment (martinholters) [inline src/unwrap.jl:12]: This was done in 0.5.0.

**Comment by galenlynch** [comment]
I think we should remove anything that was marked as deprecated in v0.6.7 or before. Seems like that's what [SemVer suggests](https://semver.org/#how-should-i-handle-deprecating-functionality), with the translation of minor versions to patch versions since we are not yet v1.

**Comment by martinholters** [comment]
...meaning you are in favor of this PR as is? (Assuming I did not miss any deprecations hiding somewhere unexpected.)

**Comment by galenlynch** [comment]
Yep!

**Comment by martinholters** [comment]
Barring objections, I'll go ahead and merge this early next week.

**Pull request merged**